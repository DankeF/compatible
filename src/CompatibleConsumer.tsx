import { ConfigProvider } from 'antd';
import { ConfigConsumer, ConfigConsumerProps } from 'antd/lib/config-provider';

export { ConfigConsumerProps };

const MergedConfigConsumer =
  ConfigProvider?.ConfigContext?.Consumer || ConfigConsumer;

export default MergedConfigConsumer;
