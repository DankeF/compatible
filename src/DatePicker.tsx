import * as React from 'react';
import { DatePicker } from 'antd';

import warning from 'rc-util/lib/warning';

type FCProps<Component> = Component extends React.ComponentType<infer Props> ? Props : never;
type InferProps = FCProps<typeof DatePicker>;

type CompatibleDatePickerProps = InferProps & {
  /** @deprecated Please use `popupClassName` instead. */
  dropdownClassName?: string;
  popupClassName?: string;
}

const CompatibleDatePicker = React.forwardRef(
  (
    { dropdownClassName, popupClassName,...restProps }: CompatibleDatePickerProps,
      ref
  ) => {
    warning(
      !dropdownClassName,
      "[antd: DatePicker] `dropdownClassName` is removed in v5, please use `popupClassName` instead.",
    );

    return (
      <DatePicker {...restProps} popupClassName={popupClassName || dropdownClassName} ref={ref} />
    );
  },
);


CompatibleDatePicker.RangePicker = React.forwardRef(
  (
    { dropdownClassName, popupClassName,...restProps }: CompatibleDatePickerProps,
    ref
  ) => {
    console.log('dropdownClassName--->>>',dropdownClassName)
    warning(
      !dropdownClassName,
      "[antd: RangePicker] `dropdownClassName` is removed in v5, please use `popupClassName` instead.",
    );

    return (
      <DatePicker.RangePicker {...restProps} popupClassName={popupClassName || dropdownClassName} ref={ref} />
    );
  },
);

if (process.env.NODE_ENV !== 'production') {
  CompatibleDatePicker.displayName = 'CompatibleDatePicker';
}

export default CompatibleDatePicker;
