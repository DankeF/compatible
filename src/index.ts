export { default as Icon } from './icon';
export { default as Form } from './form';
