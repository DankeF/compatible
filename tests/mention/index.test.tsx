import React from 'react';
import { render } from 'enzyme';
import { Mention } from '../../src';

jest.mock('draft-js/lib/generateRandomKey', () => () => '123');

describe('Mention', () => {
  it('should render correctly', () => {
    const MentionWrapper = (
      <Mention
        style={{ width: '100%' }}
        defaultSuggestions={[
          'afc163',
          'benjycui',
          'yiminghe',
          'RaoHai',
          '中文',
          'にほんご',
        ]}
      />
    );

    const wrapper = render(MentionWrapper);
    expect(wrapper).toMatchSnapshot();
  });
});
