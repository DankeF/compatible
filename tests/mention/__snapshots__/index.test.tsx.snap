// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mention should render correctly 1`] = `
<div
  class="ant-mention-wrapper"
  style="width:100%"
>
  <div
    class="ant-mention-editor oneline"
    style="width:100%"
  >
    <div
      class="ant-mention-toolbar"
    />
    <div
      class="ant-mention-editor-wrapper"
      style="width:100%"
    >
      <div
        class="DraftEditor-root"
      >
        <div
          class="DraftEditor-editorContainer"
        >
          <div
            aria-describedby="placeholder-123"
            class="notranslate public-DraftEditor-content"
            contenteditable="true"
            role="textbox"
            spellcheck="false"
            style="outline:none;user-select:text;-webkit-user-select:text;white-space:pre-wrap;word-wrap:break-word"
          >
            <div
              data-contents="true"
            >
              <div
                class=""
                data-block="true"
                data-editor="123"
                data-offset-key="123-0-0"
              >
                <div
                  class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                  data-offset-key="123-0-0"
                >
                  <span
                    data-offset-key="123-0-0"
                  >
                    <br
                      data-text="true"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
